cpp/src/Slice/Parser.cpp

```diff
@@ -1345,6 +1312,159 @@ Slice::Container::createClassDecl(const string& name, bool intf)
+                    def->updateIncludeLevel();
+                    return def;
+                }
+
+                string msg = "redefinition of interface `" + name + "'";
+                _unit->error(msg);
+            }
+        }
+        else if (differsOnlyInCase)
+        {
+            string msg = "interface definition `" + name + "' differs only in capitalization from ";
+            msg += matches.front()->kindOf() + " name `" + matches.front()->name() + "'";
+            _unit->error(msg);
+        }
+        else
+        {
+            bool declared = ClassDeclPtr::dynamicCast(matches.front());
+            string msg = "interface `" + name + "' was previously " + (declared ? "declared" : "defined")
+                + " as " + prependA(matches.front()->kindOf());
+            _unit->error(msg);
+        }
+        return 0;
+    }
+
+    bool isIllegal = !checkIdentifier(name);
+    isIllegal |= !checkForGlobalDef(name, "interface");
+    if (isIllegal)
+    {
+        return 0;
+    }
+
+    InterfaceDecl::checkBasesAreLegal(name, bases, _unit);
+
+    InterfaceDefPtr def = new InterfaceDef(this, name, bases);
+    _contents.push_back(def);
+
+    for (const auto& q : matches)
+    {
+        InterfaceDeclPtr decl = InterfaceDeclPtr::dynamicCast(q);
+        decl->_definition = def;
+    }
+
+    //
+    // Implicitly create an interface declaration for each interface
+    // definition. This way the code generator can rely on always
+    // having an interface declaration available for lookup.
+    //
+    InterfaceDeclPtr decl = createInterfaceDecl(name);
+    def->_declaration = decl;
+
+    return def;
+}
+
+InterfaceDeclPtr
+Slice::Container::createInterfaceDecl(const string& name)
+{
+    InterfaceDefPtr def;
+
+    ContainedList matches = _unit->findContents(thisScope() + name);
+    for (const auto& p : matches)
+    {
+        InterfaceDefPtr interfaceDef = InterfaceDefPtr::dynamicCast(p);
+        if (interfaceDef)
+        {
+            continue;
+        }
+
+        InterfaceDeclPtr interfaceDecl = InterfaceDeclPtr::dynamicCast(p);
+        if (interfaceDecl)
+        {
+            continue;
+        }
+
+        bool differsOnlyInCase = matches.front()->name() != name;
+        if (differsOnlyInCase)
+        {
+            string msg = "interface declaration `" + name + "' differs only in capitalization from ";
+            msg += matches.front()->kindOf() + " name `" + matches.front()->name() + "'";
+            _unit->error(msg);
+        }
+        else
+        {
+            bool declared = ClassDeclPtr::dynamicCast(matches.front());
+            string msg = "interface `" + name + "' was previously " + (declared ? "declared" : "defined")
+                + " as " + prependA(matches.front()->kindOf());
+            _unit->error(msg);
+        }
+        return 0;
+    }
+
+    bool isIllegal = !checkIdentifier(name);
+    isIllegal |= !checkForGlobalDef(name, "interface");
+    if (isIllegal)
```

Comment:
Why not this? Like we do everywhere else:
```suggestion
    if (!checkIdentifier(name) || !checkForGlobalDef(name, "interface"))
```

Reply:
Fixed